crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -523,72 +523,94 @@ impl Context {
     fn handle_array_operation(
         &mut self,
         instruction: InstructionId,
-        array: ValueId,
-        index: ValueId,
-        store_value: Option<ValueId>,
         dfg: &DataFlowGraph,
         last_array_uses: &HashMap<ValueId, InstructionId>,
     ) -> Result<(), RuntimeError> {
-        let index_const = dfg.get_numeric_constant(index);
-
-        match self.convert_value(array, dfg) {
-            AcirValue::Var(acir_var, _) => {
-                return Err(RuntimeError::InternalError(InternalError::UnExpected {
-                    expected: "an array value".to_string(),
-                    found: format!("{acir_var:?}"),
+        // Pass the instruction between array methods rather than the internal fields themselves
```

Comment:
Do you mind splitting up `handle_array_operation` some? It is getting rather long.